### ISSUE TITLE
GitVersionTask is generating an unexpectedly high version number

### ISSUE THREAD

**Issue by asbjornu** [GitTools/GitVersion#443]
As requested in shawnmclean/Mandrill-dotnet#98, I've tried to add GitVersionTask to make versioning in the project more stringent and predictable. Before adding GitVersionTask (version 3 beta 2), I created a `develop` branch and then a `feature/git-version` branch, which is the branch I've added it in. When I build that branch, the version number for some reason ends up as `4.0.1.0`. The full `GitVersionInformation` class is as following:

```csharp
[System.Runtime.CompilerServices.CompilerGenerated]
static class GitVersionInformation
{
    public static string Major = "4";
    public static string Minor = "0";
    public static string Patch = "1";
    public static string PreReleaseTag = "git-version.1";
    public static string PreReleaseTagWithDash = "-git-version.1";
    public static string BuildMetaData = "67";
    public static string FullBuildMetaData = "67.Branch.feature/git-version.Sha.5fdf5674d81fe68fa1c9496e0a128b8e1f6ed15b";
    public static string MajorMinorPatch = "4.0.1";
    public static string SemVer = "4.0.1-git-version.1";
    public static string LegacySemVer = "4.0.1-git-version1";
    public static string LegacySemVerPadded = "4.0.1-git-version0001";
    public static string AssemblySemVer = "4.0.1.0";
    public static string FullSemVer = "4.0.1-git-version.1+67";
    public static string InformationalVersion = "4.0.1-git-version.1+67.Branch.feature/git-version.Sha.5fdf5674d81fe68fa1c9496e0a128b8e1f6ed15b";
    public static string BranchName = "feature/git-version";
    public static string Sha = "5fdf5674d81fe68fa1c9496e0a128b8e1f6ed15b";
    public static string NuGetVersionV2 = "4.0.1-git-version0001";
    public static string NuGetVersion = "4.0.1-git-version0001";

}
```

The last tag on the `master` branch is `v2.0.1`. I'd expect the version of a feature branch to be `2.1.*` or something similar and at most `3.*`, so I wonder where those extra two major versions come from. Ideas?

**Comment by JakeGinnivan** [comment]
Could you post the logs for this, it should tell you where the high version is coming from

**Comment by asbjornu** [comment]
@JakeGinnivan How do I get verbose logs with GitVersionTask when building in Visual Studio?

**Comment by JakeGinnivan** [comment]
Unsure, @SimonCropp how can we get info's showing?

**Comment by asbjornu** [comment]
@JakeGinnivan Upgrading to beta 4 and building with MSBuild makes it clear what the problem is:
```
UpdateAssemblyInfo:
  INFO [07/09/15 11:54:37:32] Version not in cache. Calculating version.
  INFO [07/09/15 11:54:37:32] Project root is: C:\Users\aul\Dev\Misc\Mandrill-dotnet
MSBUILD : warning : WARN [07/09/15 11:54:37:33] No specific commit specified or found, falling back to latest commit on specified branch [C:\Users\aul\Dev\Misc\Mandrill-dotnet\src\Mandrill\Mandrill.
csproj]
  INFO [07/09/15 11:54:37:34] Begin: Attempting to inherit branch configuration from parent branch
    INFO [07/09/15 11:54:37:35] Begin: Finding branch source
    INFO [07/09/15 11:54:37:37] End: Finding branch source (Took: 22.96ms)
    INFO [07/09/15 11:54:37:43] Found possible parent branches: develop
  INFO [07/09/15 11:54:37:43] End: Attempting to inherit branch configuration from parent branch (Took: 92.79ms)
  INFO [07/09/15 11:54:37:44] Running against branch: feature/git-version (99d2975ec2df8d045cab5a1d8b76b2d63f18c072)
  INFO [07/09/15 11:54:37:45] Begin: Calculating base versions
    INFO [07/09/15 11:54:37:47] Fallback base version: 0.1.0 with commit count source 9b58d109db9cdacc0d0da6c498b352eb074623af
    INFO [07/09/15 11:54:37:50] Git tag 'v2.0.1': 2.0.1 with commit count source 9c62bd70077056baf213c9a81a8a4f05f0cdce41
    INFO [07/09/15 11:54:37:50] Git tag 'net-4.0': 4.0.0 with commit count source 655023323db1a0886b4842d8b5d40132ad8bc573
    INFO [07/09/15 11:54:37:50] Git tag 'v1.2.7': 1.2.7 with commit count source 8f55e979a169378c330ef318089732882dfab603
    INFO [07/09/15 11:54:37:50] Git tag 'v1.0.74': 1.0.74 with commit count source 0b169c844c932e168671443a4df839899f92fe91
    INFO [07/09/15 11:54:37:50] Git tag 'v1.0.66': 1.0.66 with commit count source 2ced85fad5feb21d180a3290b22782282ac93d61
    INFO [07/09/15 11:54:37:50] Git tag 'v1.0.62': 1.0.62 with commit count source cfa08cd043304e6dfba33e96f16394cd2085b51f
    INFO [07/09/15 11:54:37:50] Git tag 'build-1.0.54': 1.0.54 with commit count source 436b0f78fba3b4f1e4a5064c2676929b684f1114
    INFO [07/09/15 11:54:37:50] Git tag 'v1.0.54': 1.0.54 with commit count source 436b0f78fba3b4f1e4a5064c2676929b684f1114
    INFO [07/09/15 11:54:37:50] Git tag 'v0.1.1.4': 0.1.1+4 with commit count source c981b9e3d09fa1d2da70f90626637bf494902d14
    INFO [07/09/15 11:54:37:55] Found multiple base versions which will produce the same SemVer (4.1.0), taking oldest source for commit counting (Git tag 'net-4.0')
    INFO [07/09/15 11:54:37:55] Base version used: Git tag 'net-4.0': 4.0.0 with commit count source 655023323db1a0886b4842d8b5d40132ad8bc573
  INFO [07/09/15 11:54:37:55] End: Calculating base versions (Took: 104.78ms)
  INFO [07/09/15 11:54:37:55] Using branch name to calculate version tag
  INFO [07/09/15 11:54:37:57] 84 commits found between 655023323db1a0886b4842d8b5d40132ad8bc573 and 99d2975ec2df8d045cab5a1d8b76b2d63f18c072
```
Specifically:
```
INFO [07/09/15 11:54:37:50] Git tag 'net-4.0': 4.0.0 with commit count source 655023323db1a0886b4842d8b5d40132ad8bc573
```
Do I need to delete the `net-4.0` tag or can I instruct GitVersionTask to use the latest version number tag  `v2.0.1` as basis for the version number somehow?

**Comment by JakeGinnivan** [comment]
@asbjornu see discussion at https://github.com/GitTools/GitVersion/pull/470#issuecomment-115912417

Please weigh in, not sure how we will solve it yet.

**Comment by asbjornu** [comment]
@JakeGinnivan I think #470 will do just fine. I'm wondering why it is closed without any implementation, though?

**Comment by JakeGinnivan** [comment]
Closing this, tracking the fix in #538

**Comment by mkoertgen** [comment]
I tried to see if i could add a failing test for this specific issue with a mismatched version in a tag. 

I suspected there should be a concrete fixture covering [TaggedCommitVersionStrategy](https://github.com/GitTools/GitVersion/blob/master/src/GitVersionCore/VersionCalculation/BaseVersionCalculators/HighestTagBaseVersionStrategy.cs)

However, It seems there is no such fixture, i.e. the strategy is covered indirectly by other fixtures.
